java-frontend/src/main/java/org/sonar/java/xml/maven/PomCheckContext.java

```diff
@@ -37,11 +40,25 @@
 
   class Location {
     public final String msg;
+    @Nullable
     public final LocatedTree tree;
+    @Nullable
+    public final Integer line;
 
     public Location(String msg, LocatedTree tree) {
       this.msg = msg;
       this.tree = tree;
+      this.line = null;
+    }
+
+    public Location(String msg, int line) {
+      this.msg = msg;
+      this.tree = null;
+      this.line = line;
+    }
+
+    public boolean onLine() {
```

Comment:
As mentioned above, instead of this method, you can have method giving you the line number either from tree or the line